src/nerve_tools/configure_nerve.py

```diff
@@ -75,7 +82,7 @@ def get_labels_by_service_and_port(
     service_name: str,
     port: int,
     labels_dir: str = DEFAULT_LABEL_DIR,
-) -> Mapping[str, str]:
+) -> Dict[str, str]:
```

Comment:
Please use `MutableMapping` if you're returning dict-like object which is changeable.

Reply:
The [docs](https://docs.python.org/3/library/typing.html#typing.Dict) seem to indicate that Dict is used for return types. We know that it isn't a generic Mapping, it is in fact a Dict.

Reply:
While the doc indeed indicates that, we probably shouldn't see the doc as some kind of holy text.
Instead of that, can we please look at the situation in general and discuss a bit about *why* the doc says that.
Returning a type instead of an abstract interface is the easiest approach, as you basically allow the caller to do anything with the returned object which the type allows to be done.

But, in some cases it's better to manifest a list of actions which, in the author opinion, reasonable to do with the returned object.
That could be done by returning an interface describing allowed actions.

In our case we're returning a read-only dictionary which isn't supposed to be changed by the caller.
We can announce that by returning read-only `Mapping` instead of returning mutable `Dict` (which implements `Mutable Mapping`).
By doing that we free ourself from thinking about lifecycle of the returned object outside of our function and could do e.g. caching without analysing the code of the whole universe of callers of our function.

In general, it's a good practice to restrict types and make them as restrictive and specific as possible.